### PR TITLE
Edit and delete tweet

### PR DESCRIPTION
**Pull request by GaganDureja**

### CODE REVIEW

File: templates/home.html [hunk 1]
```diff
@@ -34,6 +34,10 @@ <h4>All Tweets</h4>
     <h6>{{ tweet.user.first_name }}</h6>
     <p>{{ tweet.tweet_message }}</p>
     <a href="{% url 'tweets:showTweet' tweet.id %}">Show Tweet</a>
+    {% if request.user ==  tweet.user %}
```
Comment (grsahil20) [inline templates/home.html:37]: Use
```suggestion
    {% if request.user.id ==  tweet.user_id %}
```

saves one query

File: tweets/views.py [hunk 2]
```diff
@@ -12,4 +12,30 @@ def createTweet(request):
 
 def showTweet(request,id):
   tweet_details = get_object_or_404(Tweet, id=id)
-  return render(request, 'tweets/show.html',{'tweet_details':tweet_details})
\ No newline at end of file
+  return render(request, 'tweets/show.html',{'tweet_details':tweet_details})
+
+def editTweet(request,id):
+  tweet_details = get_object_or_404(Tweet, id=id)
```
Comment (grsahil20) [inline tweets/views.py:18]: Why not put request.user in query and only find the ones created by user

File: tweets/views.py [hunk 3]
```diff
@@ -12,4 +12,30 @@ def createTweet(request):
...
+  if request.user != tweet_details.user:
+    messages.error(request, "Invalid path")
+    return redirect('home')
+  return render(request, 'tweets/edit.html',{'tweet_details':tweet_details})
+
+def updateTweet(request,id):
+  tweet_details = get_object_or_404(Tweet, id=id)
```
Comment (grsahil20) [inline tweets/views.py:25]: Also here

File: tweets/views.py [hunk 4]
```diff
@@ -12,4 +12,30 @@ def createTweet(request):
...
+  if request.user != tweet_details.user:
+    messages.error(request, "Invalid path")
+    return redirect('home')
+  tweet_details.tweet_message = request.POST.get('tweet_message')
+  tweet_details.save()
+  messages.success(request, "Tweet updated")
+  return redirect('home')
+
+def destroyTweet(request,id):
+  tweet_details = get_object_or_404(Tweet, id=id)
```
Comment (grsahil20) [inline tweets/views.py:35]: And here

File: tweets/views.py (hunk 2 above)
Comment (GaganDureja) [reply tweets/views.py:18]: This can be done

File: templates/home.html (hunk 1 above)
Comment (GaganDureja) [reply templates/home.html:37]: I didn't used any query to compare. Maybe django have some inbuilt function to check.

File: tweets/views.py (hunk 2 above)
Comment (GaganDureja) [reply tweets/views.py:18]: I've done to 
```python
tweet_details = get_object_or_404(Tweet, id=id, user_id=request.user.id)
```

File: templates/home.html (hunk 1 above)
Comment (grsahil20) [reply templates/home.html]: how do you think Django with know who is the User for Tweet

Comment (GaganDureja) [reply templates/home.html]: yes there must be something, but this is why we are using framework that we don't want to write large code and use inbuilt functions. I learned about this so I used it as I know and it was working fine.

File: templates/home.html [hunk 5]
```diff
@@ -34,6 +34,10 @@ <h4>All Tweets</h4>
     <h6>{{ tweet.user.first_name }}</h6>
     <p>{{ tweet.tweet_message }}</p>
     <a href="{% url 'tweets:showTweet' tweet.id %}">Show Tweet</a>
+    {% if request.user.id ==  tweet.user_id %}
```
[the remark this answers is not in the document]
Comment (GaganDureja) [reply templates/home.html:37]: yes its working

File: tweets/views.py [hunk 6]
```diff
@@ -12,4 +12,21 @@ def createTweet(request):
 
 def showTweet(request,id):
   tweet_details = get_object_or_404(Tweet, id=id)
-  return render(request, 'tweets/show.html',{'tweet_details':tweet_details})
\ No newline at end of file
+  return render(request, 'tweets/show.html',{'tweet_details':tweet_details})
+
+def editTweet(request,id):
+  tweet_details = get_object_or_404(Tweet, id=id, user_id=request.user.id)
```
[the remark this answers is not in the document]
Comment (grsahil20) [reply tweets/views.py]: No, tweet is more specific. 
You can add `details` to user_details, tag_details, category_details. Doesn't help.

**Review by grsahil20 — APPROVED** [review]

**Pull request merged**